content/Improve your Rails App searches with Elasticsearch - Leandro Dominguez.md

```diff
@@ -0,0 +1,213 @@
+
+Sometimes (for performance or even business requirements), you don’t want to update data in Elasticsearch instantaneously every time a record has been changed in our main database (this is customizable but we won’t tackle this in the current article).
+
+**Step 3: Defining our Elasticsearch index**
+
+In Elasticsearch we don’t have tables or rows. Our models are stored in ‘_indexes’_ and each instance is called ‘_document’_. Although Elastic can automatically detect the attributes types of our model, we are going to declare them explicitly so we have more control over them by adding the following to your model:
+
+```ruby
+ settings do
+   mappings dynamic: false do
+     indexes :id, index: :not_analyzed
+     indexes :title, type: :text, analyzer: :english
+     indexes :body, type: :text, analyzer: :english
+     indexes :topic, type: :keyword
+   end
+ end
+```
+
+What are we doing here?
+
+In the third line we are telling Elastic we want to take our `id` attribute and add it to the `Post`’ index as an index but without analyzing it. This means it won’t apply any filter or transformation to it.
+
+In the fourth and fifth line, we’re saying we are adding both `title` and `body` as text type and with an English analyzer. This will automatically run text tokenizers, analyzers and will apply a lot of natural language processing tools every time we insert or edit a document in our index, making the searches faster, easier and typo-resistant (be aware all these things take an extra time). Follow these links if you’re interested in learning more about [language analyzers]([https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-lang-analyzer.html](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-lang-analyzer.html)) and [tokenizers]([https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-tokenizers.html](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-tokenizers.html))
+
+And last but not least, we are declaring `topic` as a _keyword_, meaning we are expecting this field only to be tokenized (and not processed or analyzed) making the inserts and edits faster. This is because we are not expected to search through this field, but feel free to play with different options.
+
+**Step 4: Starting our Elasticsearch instance and creating our index**
+
+Assuming you already installed Elasticsearch, start an instance of Elastic by just executing in a terminal **elasticsearch** (remember I’m using _homebrew_). This will launch a server in the 9200 port by default. To check if everything is working as expected make a get request to [http://localhost:9200/](http://localhost:9200/) (could be just opening the URL in a browser).
```

Comment:
While using homebrew shouldn't you do `brew services start elasticsearch`? Just a smartass comment to avoid other smartasses, I'm sure both work

Reply:
Only elasticsearch works for me 🤔

Reply:
Yeah it was just to use the preferred way of running it